src/Serilog/Context/LogContext.cs

```diff
@@ -75,37 +76,80 @@ public static class LogContext
         /// <returns>A token that must be disposed, in order, to pop properties back off the stack.</returns>
         public static IDisposable PushProperty(string name, object value, bool destructureObjects = false)
         {
+            return Push(new PropertyEnricher(name, value, destructureObjects));
+        }
+
+        /// <summary>
+        /// Push an enricher onto context, returning an <see cref="IDisposable"/>
+        /// that must later be used to remove the property, along with any others that
+        /// may have been pushed on top of it and not yet popped. The property must
+        /// be popped from the same thread/logical call context.
+        /// </summary>
+        /// <param name="enricher">Log Properties to push onto the log context</param>
```

Comment:
why "Log Properties" (capitalised) ? I notice this was just copied, but maybe now is a good time to clean it up?